game/board.py

```diff
@@ -117,10 +132,18 @@ def _gold_positions(self) -> 'list[tuple]':
             if self._board[row][col].gold > 0
         ]
 
-    def _player_can_find_gold(self, row, col, golds):
+    def _player_can_find_gold(
+        self,
+        player_position: PlayerPosition,
+        golds,
+    ) -> bool:
         for gold_position in golds:
-            if not self._can_find_gold(row, col, gold_position, []):
-                self._board[row][col].has_hole = False
+            if not self._can_find_gold(
+                player_position.row,
+                player_position.col,
```

Comment:
In future PR's think of changing `_can_find_gold` arguments `player_position.row` and `player_position.col` for `player_position` instead

Reply:
I understand this comment. I was thinking create some tickets to refactor these things to make the code more readable